pkg/o-neko-catnip/oneko/oneko.go

```diff
@@ -30,9 +31,14 @@ func New(configuration *config.Config, ctx context.Context) *ONekoApi {
 		panic(err)
 	}
 
-	requestCache := ttlcache.NewCache()
+	var api *ONekoApi
 
-	api := &ONekoApi{
+	requestCache := ttlcache.New[string, *cacheEntry](
+		ttlcache.WithTTL[string, *cacheEntry](time.Duration(configuration.ONeko.Api.CacheRequestsInMinutes)*time.Minute),
+		ttlcache.WithDisableTouchOnHit[string, *cacheEntry](),
```

Comment:
You could pass the load function here directly using `ttlcache.WithLoader[string, *cacheEntry](o.loaderFunc)`

Reply:
Done.